frontend/src/metabase/query_builder/components/view/ViewFooter/ViewFooterControl.tsx

```diff
@@ -36,55 +37,30 @@ const ViewFooterControl = ({
   const vizIcon = getIconForVisualizationType(question.display());
   const isShowingRawTable = useSelector(getIsShowingRawTable);
   const isVisualized = useSelector(getIsVisualized);
+  const viewFooterControlState: "editor" | "results" | "visualization" =
+    useSelector(getViewFooterControlState);
 
   const dispatch = useDispatch();
 
-  const [value, setValue] = useState<"editor" | "table" | "visualization">(
-    isNotebook ? "editor" : isShowingRawTable ? "table" : "visualization",
-  );
-
-  useEffect(() => {
-    // switch from editor to view
-    if (!isNotebook && value === "editor") {
-      setValue("visualization");
-    }
-  }, [isNative, isNotebook, value]);
+  const initialValue = isNotebook
```

Comment:
initializeQB is called only once, but we need to listen to back/forth browser buttons

https://github.com/user-attachments/assets/ebd46ae8-9d72-4a07-a39e-fee13a815f2e

Do you have in mind another place where we can initialize the value? otherwise, I'd keep this logic in the component

Reply:
fixed [94e1dcc](https://github.com/metabase/metabase/pull/50498/commits/94e1dccb4797ff147a4e26b74e5e478746e5b127)